ReactiveXComponent/RabbitMq/RabbitMqSubscriber.cs

```diff
@@ -19,112 +19,128 @@ public class RabbitMqSubscriber : IXCSubscriber
-                            k.SatetMachineRef.StateMachineCode == _xcConfiguration.GetStateMachineCode(component, stateMachine));
+            InitSubscriber(component, stateMachine, _privateCommunicationIdentifier);
         }
 
         public IObservable<MessageEventArgs> GetStateMachineUpdates()
         {
             return _xcObservable;
         }
 
-        private void InitSubscriber(string component, string stateMachine)
+        private void InitSubscriber(string component, string stateMachine, string privateCommunicationIdentifier = null)
         {
-            var exchangeName = _xcConfiguration?.GetComponentCode(component).ToString();
-            var routingKey = _xcConfiguration?.GetSubscriberTopic(component, stateMachine);
+            if (_xcConfiguration == null) return;
+            var exchangeName = _xcConfiguration.GetComponentCode(component).ToString();
+            var routingKey = string.IsNullOrEmpty(privateCommunicationIdentifier) ? _xcConfiguration.GetSubscriberTopic(component, stateMachine) : privateCommunicationIdentifier;
             try
             {
-                if (_connection == null || !_connection.IsOpen) return;
-                var channel = _connection.CreateModel();
-
-                channel.ModelShutdown += ChannelOnModelShutdown;
-                channel.ExchangeDeclare(exchangeName, ExchangeType.Topic);
-                var queueName = channel.QueueDeclare().QueueName;
-                var subscriber = new EventingBasicConsumer(channel);
-                channel.BasicConsume(queueName, false, subscriber);
-                channel.QueueBind(queueName, exchangeName, routingKey, null);
-
+                IModel channel;
+                EventingBasicConsumer subscriber;
+                CreateExchangeChannel(exchangeName, routingKey, out channel, out subscriber);
+                if (channel == null || subscriber == null) return;
                 var rabbitMqSubscriberInfos = new RabbitMqSubscriberInfos
                 {
                     Channel = channel,
                     Subscriber = subscriber
                 };
-                var subscriberKey = new SubscriberKey(Convert.ToInt32(exchangeName),
-                    _xcConfiguration.GetStateMachineCode(component, stateMachine));
 
-                if (!_subscribers.ContainsKey(subscriberKey))
-                {
-                    _subscribers.Add(subscriberKey, rabbitMqSubscriberInfos);
-                }
-                ReceiveMessage(rabbitMqSubscriberInfos);
+                var subscriberKey = new SubscriberKey(_xcConfiguration.GetComponentCode(component), _xcConfiguration.GetStateMachineCode(component, stateMachine));
+                _subscribers.AddOrUpdate(subscriberKey, rabbitMqSubscriberInfos, (key, oldValue) => rabbitMqSubscriberInfos);
+                ReceiveMessage(subscriberKey);
             }
             catch (OperationInterruptedException e)
```

Comment:
You caught a specific exception then threw a general one. You should create a new Exception class called something like _RXCException_ and always throw that type of exception evey time you want to throw one. This way a ReactiveXComponent user would have to catch one type of exceptions.